### PR TITLE
Feat: Store negative/inverted types in type propagator nodes

### LINKED ISSUE #4: Store negative/inverted types in type propagator nodes
# Feature or enhancement

Type propagator nodes should be able to store that it's *not* a certain type, eg. `!int` or `!float`.

# Pitch

This will:
- simplify the type prop state
- prevent duplicate failed type checks, e.g. if we know something is `!int`, we don't need to check it's an `int` again. The current lbbv paper which uses only positive type success information can't do this.

### PR DESCRIPTION
**Pull request by JuliaPoo**
This removes the ladder guard entirely.
Closes #4 and #5.

### CODE REVIEW

File: Python/tier2.c [hunk 1]
```diff
@@ -36,6 +36,83 @@ PyTypeObject PySmallInt_Type = {
...
 static inline int IS_SCOPE_EXIT_OPCODE(int opcode);
 
+////////// TYPE NODES FUNCTIONS
+
+/**
+ * @brief Maps _Py_NegativeTypeMaskBit to typeobject
+ * @param bitidx _Py_NegativeTypeMaskBit
+ * @return Corresponding typeobject
+*/
+PyTypeObject *bit_to_typeobject(int bitidx)
```
Comment (Fidget-Spinner) [inline Python/tier2.c]: PEP 7 nits, same for all the new functions below
```suggestion
PyTypeObject *
bit_to_typeobject(int bitidx)
```

**Review by Fidget-Spinner — COMMENTED** [review]
just stylistic nits, otherwise LGTM. Thanks.

File: Python/tier2.c (hunk 1 above)
Comment (JuliaPoo) [reply Python/tier2.c]: Okie done

**Review by Fidget-Spinner — APPROVED** [review]

**Comment by JuliaPoo** [comment]
Oh yea, also closes #26

**Pull request merged**